src/main/java/org/tikv/common/region/UnreachableStoreChecker.java

```diff
@@ -0,0 +1,77 @@
+package org.tikv.common.region;
+
+import io.grpc.ManagedChannel;
+import io.grpc.health.v1.HealthCheckRequest;
+import io.grpc.health.v1.HealthCheckResponse;
+import io.grpc.health.v1.HealthGrpc;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import org.tikv.common.ReadOnlyPDClient;
+import org.tikv.common.util.ChannelFactory;
+
+public class UnreachableStoreChecker implements Callable<Object> {
+  private ConcurrentHashMap<Long, TiStore> stores;
+  private List<TiStore> taskQueue;
+  private final ChannelFactory channelFactory;
+  private final ReadOnlyPDClient pdClient;
+
+  public UnreachableStoreChecker(ChannelFactory channelFactory, ReadOnlyPDClient pdClient) {
+    this.stores = new ConcurrentHashMap();
+    this.taskQueue = new LinkedList<>();
+    this.channelFactory = channelFactory;
+    this.pdClient = pdClient;
+  }
+
+  public void scheduleStoreHealthCheck(TiStore store) {
+    TiStore oldStore = this.stores.get(Long.valueOf(store.getId()));
+    if (oldStore != null) {
+      return;
+    }
+    synchronized (this.taskQueue) {
+      this.stores.put(Long.valueOf(store.getId()), store);
+      this.taskQueue.add(store);
+    }
```

Comment:
Is this function possible to be called concurrently? May it be a race that multiple threads do the check-and-insert-if-not-exist operation at the same time?